frontend/shared/components/src/context/hooks/useSetUitpasEvent.ts

```diff
@@ -0,0 +1,128 @@
+        || json.location.name === null
+        || !('nl' in json.location.name)
+        || typeof json.location.name.nl !== 'string'
+        || (('startDate' in json) && (typeof json.startDate !== 'string' || json.startDate === ''))
+        || ('endDate' in json && (typeof json.endDate !== 'string' || json.endDate === ''))
+    ) {
+        console.error('Invalid event response', json);
+        throw new SimpleError({
+            code: 'invalid_event_response',
+            message: `Invalid event response`,
+            human: $t(`Er is een fout opgetreden bij het ophalen van het UiTPAS-evenement.`),
+        });
+    }
+}
+
+export function useSetUitpasEvent(patchedProduct: Ref<Product>, addProductPatch: (newPatch: PartialWithoutMethods<AutoEncoderPatchType<Product>>) => void) {
+    const { getOfficialUitpasSocialTariff } = useGetOfficialUitpasSocialTariff();
+
+    const getUitpasEventDetails = async (eventUrl: string): Promise<UitpasEventResponse> => {
+        const response = await fetch(eventUrl).catch(() => {
+            throw new SimpleError({
+                code: 'fetch_error_getting_uitpas_event_details',
+                message: `Error fetching Uitpas event details`,
+                human: $t(`Er is een fout opgetreden bij het ophalen van het UiTPAS-evenement.`),
+            });
+        });
+
+        if (!response.ok) {
+            throw new SimpleError({
+                code: 'invalid_response_getting_uitpas_event_details',
+                message: `Invalid response when getting Uitpas event details`,
+                human: $t(`Er is een fout opgetreden bij het ophalen van het UiTPAS-evenement.`),
+            });
+        }
+        const json = await response.json().catch(() => {
+        // Handle JSON parsing errors
+            throw new SimpleError({
+                code: 'json_parsing_error_getting_uitpas_event_details',
+                message: `Error parsing JSON when getting Uitpas event details`,
+                human: $t(`Er is een fout opgetreden bij het ophalen van het UiTPAS-evenement.`),
+            });
+        });
+
+        assertIsEventResponse(json);
+        const eventResponse = new UitpasEventResponse();
+        eventResponse.url = json['@id'];
+        eventResponse.name = json.name.nl;
+        eventResponse.location = json.location.name.nl;
+        if (json.startDate) {
+            eventResponse.startDate = new Date(json.startDate);
+        }
+        if (json.endDate) {
+            eventResponse.endDate = new Date(json.endDate);
+        }
+        return eventResponse;
+    };
+
+    const setUitpasEventUrl = async (newUrl: string | null): Promise<void> => {
+        const details = newUrl ? await getUitpasEventDetails(newUrl) : null;
+
+        await setUitpasEvent(details);
+    };
+
+    const setUitpasEvent = async (uitpasEvent: UitpasEventResponse | null): Promise<void> => {
+        // store the new details
+        addProductPatch({ uitpasEvent });
```

Comment:
Het valt te overwegen om hier een await nextTick() toe te voegen omdat het in sommige gevallen `addProductPatch` via een emit werkt, en dan riskeer je onverwachte bugs. Denk dat die nu niet het geval is, maar kan in de toekomst mogelijks foutjes voorkomen

```suggestion
        addProductPatch({ uitpasEvent });
        await nextTick();
```